go/libraries/doltcore/sqle/tables.go

```diff
@@ -934,14 +958,29 @@ func parseFkReferenceOption(refOp sql.ForeignKeyReferenceOption) (doltdb.Foreign
 }
 
 // createIndex creates the given index on the given table with the given schema. Returns the updated table, updated schema, and created index.
-func createIndexForTable(ctx *sql.Context, table *doltdb.Table, indexName string, using sql.IndexUsing, constraint sql.IndexConstraint, columns []sql.IndexColumn, comment string) (*doltdb.Table, schema.Schema, schema.Index, error) {
+func createIndexForTable(
+	ctx *sql.Context,
+	table *doltdb.Table,
+	indexName string,
+	using sql.IndexUsing,
+	constraint sql.IndexConstraint,
+	columns []sql.IndexColumn,
+	isUserDefined bool,
+	comment string,
+) (
+	newTable *doltdb.Table,
```

Comment:
This return type is getting a little chonky... I would take the time to declare a return struct instead. (leave the error as a separate return result)